launcher/ui/pages/modplatform/ResourceModel.cpp

```diff
@@ -132,6 +132,36 @@ void ResourceModel::search()
     if (hasActiveSearchJob())
         return;
 
+    if (m_search_term.startsWith("#")) {
+        auto projectId = m_search_term.mid(1);
+        if (!projectId.isEmpty()) {
+            ResourceAPI::ProjectInfoCallbacks callbacks;
+
+            // Use defaults if no callbacks are set
+            if (!callbacks.on_fail)
+                callbacks.on_fail = [this](QString reason) {
+                    if (!s_running_models.constFind(this).value())
+                        return;
+                    searchRequestFailed(reason, -1);
+                };
+            if (!callbacks.on_abort)
+                callbacks.on_abort = [this] {
+                    if (!s_running_models.constFind(this).value())
+                        return;
+                    searchRequestAborted();
+                };
+
+            if (!callbacks.on_succeed)
+                callbacks.on_succeed = [this](auto& doc, auto& pack) {
+                    if (!s_running_models.constFind(this).value())
+                        return;
+                    searchRequestForOneSucceeded(doc);
+                };
```

Comment:
why
the bodies of these if statements will always run, right?

Reply:
yeah you are right(copied from other place where the arguments are not guaranteed to be empty)
fixed now